code/modules/reagents/Chemistry-Recipes.dm

```diff
@@ -3820,12 +3784,73 @@
 	if(!location)
 		location = get_turf(holder.my_atom)
 	for(var/i=1 to created_volume)
-		new /mob/living/simple_animal/hostile/carp/baby(location)
+		if(prob(80))
+			new mob2spawnA(location)
```

Comment:
you're almost there, instead of doing it like this, just replace this whole if block with a call to a new proc, name it whatever you want like `reagent_mob_spawn()`

then you can make each chem spawn its specific number of things easily, like 3 cockroaches or 5 mice or whatever, you don't have to make them all uniform

Reply:
That if block is just so aminoblatella can spawn its 2 kinds of mobs correctly (since the other hostilesynth with this structure makes exclusively carp babies): 80% chance to make a mutated cockroach, 20% chance to make a matriarch if mixed with mutagen.
And if you want more mobs per reaction, you can just increase the "amount" of the recipe accordingly, see the aminoblatella+nutriment reaction, it makes 3 regular cockroaches. Tested.
Can't it stay like this for now? I don't see yet another amino- chem happening anytime soon.

Reply:
Okay, that's fine then. The other way would be cleaner but this is a step better than it was.